blueflood-http/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/HttpMultiRollupsQueryHandler.java

```diff
@@ -111,14 +111,15 @@ public void handle(ChannelHandlerContext ctx, HttpRequest request) {
             return;
         }
 
-        HTTPRequestWithDecodedQueryParams requestWithParams = (HTTPRequestWithDecodedQueryParams) request;
+        List<String> locatorStrings = new ArrayList<String>();
+        for (Locator locator : locators)
+            locatorStrings.add(locator.toString());
 
+        HTTPRequestWithDecodedQueryParams requestWithParams = (HTTPRequestWithDecodedQueryParams) request;
         final Timer.Context httpBatchMetricsFetchTimerContext = httpBatchMetricsFetchTimer.time();
         try {
             RollupsQueryParams params = PlotRequestParser.parseParams(requestWithParams.getQueryParams());
-            BatchMetricsQuery query = new BatchMetricsQuery(locators, params.getRange(), params.getGranularity());
-            Map<Locator, MetricData> results = new BatchMetricsQueryHandler(executor, AstyanaxReader.getInstance())
-                                                        .execute(query, queryTimeout);
+            Map<Locator, MetricData> results = getRollupByGranularity(tenantId, locatorStrings, params.getRange().getStart(), params.getRange().getStop(), params.getGranularity());
```

Comment:
I don't think we plot these currently, but I see your point. I will add them to the `HttpMultiplotHandler` code.